pytorch_lightning/loggers/neptune.py

```diff
@@ -11,29 +12,80 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+#
 """
 Neptune Logger
 --------------
 """
+__all__ = [
+    "NeptuneLogger",
+]
+
 import logging
+import operator
+import os
 from argparse import Namespace
-from typing import Any, Dict, Iterable, Optional, Union
+from functools import reduce
+from typing import Any, Dict, Optional, Set, Union
+from weakref import ReferenceType
 
 import torch
-from torch import is_tensor
 
+from pytorch_lightning import __version__
 from pytorch_lightning.loggers.base import LightningLoggerBase, rank_zero_experiment
 from pytorch_lightning.utilities import _module_available, rank_zero_only
+from pytorch_lightning.utilities.imports import _compare_version
+from pytorch_lightning.utilities.model_summary import ModelSummary
+from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
 
-log = logging.getLogger(__name__)
 _NEPTUNE_AVAILABLE = _module_available("neptune")
-
-if _NEPTUNE_AVAILABLE:
-    import neptune
-    from neptune.experiments import Experiment
+_NEPTUNE_GREATER_EQUAL_0_9 = _NEPTUNE_AVAILABLE and _compare_version("neptune", operator.ge, "0.9.0")
+
+if _NEPTUNE_AVAILABLE and _NEPTUNE_GREATER_EQUAL_0_9:
+    try:
+        from neptune import new as neptune
+        from neptune.new.run import Run
+        from neptune.new.exceptions import NeptuneLegacyProjectException
+    except ImportError:
+        import neptune
+        from neptune.run import Run
+        from neptune.exceptions import NeptuneLegacyProjectException
 else:
-    # needed for test mocks, these tests shall be updated
-    neptune, Experiment = None, None
+    # needed for test mocks, and function signatures
+    neptune, Run = None, None
+
+
+log = logging.getLogger(__name__)
+
+INTEGRATION_VERSION_KEY = "source_code/integrations/pytorch-lightning"
+
+# kwargs used in NeptuneLogger for legacy client (current NeptuneLegacyLogger)
+LEGACY_NEPTUNE_INIT_KWARGS = [
```

Comment:
Sorry I don't get this, this is list of arguments which were available in previous `NeptuneLogger` and are not available in current `NeptuneLogger`.
Where should I store them?

I can make it a private list of `NeptuneLogger` if you want.

Reply:
I meant, I think you could do:

```py
from neptune.pytorch_lightning import _ INTEGRATION_VERSION_KEY, _LEGACY_NEPTUNE_INIT_KWARGS, ...
```

Reply:
I see.
But I don't really think that it's good idea to keep those variables in separate repository.
We'd have to update our `neptune-client`, release it and then add proper version as minimal supported dependency to `pytorch-lightning` library.
It could make some sense if `LEGACY_NEPTUNE_INIT_KWARGS` would be used in several places, then `neptune.pytorch_lightning` would be the source of true.
But this list keeps difference between args accepted by previous version of `NeptuneLogger` and current version of `NeptuneLogger` I don't see any reason why should we keep it in other repo.

`INTEGRATION_VERSION_KEY` is something we always keep as global variable in our integrations code.

I agree that we could make those variables "module private" by prefixing them with `"_"`, or add `__all__` statement which will make only `NeptuneLogger` importable.

Reply:
@tchaton what do you think?